packages/core/src/codewhisperer/service/inlineCompletionService.ts

```diff
@@ -137,6 +137,9 @@ export class InlineCompletionService {
                 if (RecommendationHandler.instance.checkAndResetCancellationTokens()) {
                     RecommendationHandler.instance.reportUserDecisions(-1)
                     await vscode.commands.executeCommand('aws.codeWhisperer.refreshStatusBar')
+                    if (triggerType === 'OnDemand' && session.recommendations.length === 0) {
```

Comment:
This code path does not include SM studio & C9. Do you want to include them?

Reply:
I would like to follow up that in another PR

Reply:
the old one is located in `recommendationHandler.ts` https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/codewhisperer/service/recommendationHandler.ts , which should handle different IDEs. It take more time to finish an fix-all approach. Let's treat this as a hot fix.